libopae/plugins/xfpga/token_list.c

```diff
@@ -160,23 +178,69 @@ struct _fpga_token *token_get_parent(struct _fpga_token *_t)
 {
 	char *p;
 	char spath[SYSFS_PATH_MAX];
-	int device_instance;
 	struct token_map *itr;
 	int err = 0;
+	errno_t e;
+	DIR *dir;
+	struct dirent *dirent;
+	int i;
+	int found;
 
 	p = strstr(_t->sysfspath, FPGA_SYSFS_AFU);
 	if (!p) // FME objects have no parent.
 		return NULL;
 
-	p = strrchr(_t->sysfspath, '.');
-	if (!p)
+	// Find the parent FME device of the specified Port device.
```

Comment:
Should we create an internal object that refers to the parent and then stash that in the fme/port object? Or at the very least, stash the path to the top device?